### PR TITLE
Clone from gitlab instead of using asp

### PR DESCRIPTION
**Pull request by kpcyrd**

### CODE REVIEW

File: repro.in [hunk 1]
```diff
@@ -377,31 +375,28 @@ function cmd_check(){
...
 
     EPHEMERAL=1 exec_nspawn root --bind="${build_root_dir}/startdir:/startdir" --bind="$(readlink -e ${cachedir}):/var/cache/pacman/pkg" \
     bash <<-__END__
 shopt -s globstar
-pacman -S asp --noconfirm --needed
-if ! asp checkout $pkgbase; then
+
+pacman -S git --noconfirm --needed
+
+if ! git clone --depth 1 --branch "$pkgver" "https://gitlab.archlinux.org/archlinux/packaging/packages/$pkgbase.git"; then
```
Comment (jelly) [inline repro.in]: This won't work as Gitlab restricts some characters in project names or even names such as `tree`. `pkgctl` has a custom lookup function for this. This is why we should use `pkgctl`

**Review by jelly — CHANGES_REQUESTED** [review]

File: repro.in (hunk 1 above)
Comment (anthraxx) [reply repro.in]: pkgctl repo clone --unprivileged --switch "${pkgver}" "${pkgbase}"

Comment (kpcyrd) [reply repro.in]: Doesn't work yet:
```
==> ERROR: invalid argument: --unprivileged
```

Comment (anthraxx) [reply repro.in]: I've put devtools `1:1.0.1-1` into `extra-testing`, please give it a shot and give back feedback if that works as expected or anything I can quickly fix as well.

It should work as follows:
`pkgctl repo clone --protocol https --switch "${pkgver}" "${pkgbase}"`

**Comment by kpcyrd** [comment]
Current error:
```
(1/3) Rebuilding certificate stores...
(2/3) Warn about old perl modules
(3/3) Updating the info directory file...
Installing devtools from https://archive.archlinux.org/packages/d/devtools-1:1.0.0-1-any.pkg.tar.zst
loading packages...
looking for conflicting packages...

Packages (1) devtools-1:1.0.0-1

Total Installed Size:  0.23 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                         [#######################################] 100%
(1/1) checking package integrity                                       [#######################################] 100%
(1/1) loading package files                                            [#######################################] 100%
(1/1) checking for file conflicts                                      [#######################################] 100%
(1/1) checking available disk space                                    [#######################################] 100%
:: Processing package changes...
(1/1) installing devtools                                              [#######################################] 100%
Optional dependencies for devtools
    btrfs-progs: btrfs support
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
cp: cannot stat '/usr/share/devtools/makepkg-x86_64.conf': No such file or directory
  -> Delete snapshot for pmbootstrap_168102...
```

Permalink to a package to reproduce this with: https://archive.archlinux.org/repos/2023/05/25/extra/os/x86_64/doctl-1.96.0-1-x86_64.pkg.tar.zst

**Comment by Foxboron** [comment]
> cp: cannot stat '/usr/share/devtools/makepkg-x86_64.conf': No such file or directory

The file has moved to `/usr/share/devtools/makepkg.conf.d/x86_64.conf`

**Comment by Foxboron** [comment]
Ah, that is very annoying as we need to special case this.

**Comment by Foxboron** [comment]
```diff
diff --git a/repro.in b/repro.in
index 7ce2528..9756428 100755
--- a/repro.in
+++ b/repro.in
@@ -497,7 +497,8 @@ if [[ "$DEVTOOLS_PKG" == https* ]]; then
 else
   pacman --noconfirm --needed -Sddu "$DEVTOOLS_PKG"
 fi
-cp -v /usr/share/devtools/makepkg-x86_64.conf /mnt/etc/makepkg.conf
+[[ -f /usr/share/devtools/makepkg-x86_64.conf ]] && cp -v /usr/share/devtools/makepkg-x86_64.conf /mnt/etc/makepkg.conf
+[[ -f /usr/share/devtools/makepkg.d/x86_64.conf ]] && cp -v /usr/share/devtools/makepkg.d/x86_64.conf /mnt/etc/makepkg.conf
 __END__
     lock_close 9 "$KEYRINGCACHE/$keyring_package.lock"
```

Should work

**Comment by kpcyrd** [comment]
I made it work with errexit and tested it with one package before the migration and one after

**Comment by Foxboron** [comment]
Awesome :) thanks!

**Review by Foxboron — APPROVED** [review]

**Pull request merged**